File: Java/jan26/counter_game.java
import java.io.*;
import java.math.*;
import java.security.*;
import java.text.*;
import java.util.*;
import java.util.concurrent.*;
import java.util.function.*;
import java.util.regex.*;
import java.util.stream.*;
import static java.util.stream.Collectors.joining;
import static java.util.stream.Collectors.toList;

class Result {
    /*
     * Complete the 'counterGame' function below.
     *
     * The function is expected to return a STRING.
     * The function accepts LONG_INTEGER n as parameter.
     */

    public static String counterGame(long n) {
    // Write your code here
    int count = 0;
        while(n!=1)
        {
            if(Result.isPower(n))
                n = n/2;
            else 
                n = n - Result.nextPower(n);
            count++;
        }
        if(count%2 == 0) return "Richard";
        else return "Louise" ;
    }
    public static boolean isPower(long n)
    {
        double x = Math.log(n)/Math.log(2);
        if(x==(int)x) return true;
        else return false;
    }
    
    public static long nextPower(long n)
    {
        long m = 1;
        while(m<n)
            m = m * 2;
        return m/2;
        
    }

}

public class Solution {
    public static void main(String[] args) throws IOException {
        BufferedReader bufferedReader = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter bufferedWriter = new BufferedWriter(new FileWriter(System.getenv("OUTPUT_PATH")));

        int t = Integer.parseInt(bufferedReader.readLine().trim());

        IntStream.range(0, t).forEach(tItr -> {
            try {
                long n = Long.parseLong(bufferedReader.readLine().trim());

                String result = Result.counterGame(n);

                bufferedWriter.write(result);
                bufferedWriter.newLine();
            } catch (IOException ex) {
                throw new RuntimeException(ex);
            }
        });

        bufferedReader.close();
        bufferedWriter.close();
    }
}

File: Python/jan26/counter_game.py
#!/bin/python3

import math
import os
import random
import re
import sys

#
# Complete the 'counterGame' function below.
#
# The function is expected to return a STRING.
# The function accepts LONG_INTEGER n as parameter.
#


def counterGame(n):
    count=0
    while n>1:
        x=math.log2(n)
        if x==int(x):
            count+=x
            break
        else:
            n-=2**(int(x))
            count+=1    

    if count%2 == 0:
        return 'Richard'
    else:
        return 'Louise'

if __name__ == '__main__':
    fptr = open(os.environ['OUTPUT_PATH'], 'w')

    t = int(input().strip())

    for t_itr in range(t):
        n = int(input().strip())

        result = counterGame(n)

        fptr.write(result + '\n')

    fptr.close()
